### PR TITLE
Evo 1237 grid view for feed

### LINKED ISSUE #1237: Grid view for the Projects/Prize Applications page
Add third view type to Stream + List, now Grid. Projects view uses this as a default

### PR DESCRIPTION
**Pull request by thomasgwatson**
Closes #1237 

I have parred down what is included in the `PostCard` and `PostRow` to the `PostGridItem`. The `PostGridItem` is a small and square space to work with, so its fiddly to display the information in the same way.

@brodeur I commented out several of the post details but they should be relatively easily uncommented to test out adding more or otherwise changing up the styling of the `PostGridItem`. Once this is done, I can come back through and clean-up what is no longer actually needed for this PR.

grid-projects-2
grid-posts-1
grid-posts-2
grid-projects-1

### CODE REVIEW

**Comment by brodeur** [comment]
[@Tom Watson](https://terrans.slack.com/team/U013GH1FLKX)
 I still have a bit more work to do on grid view but I could use a hand. I got stuck for an hour trying to figure it out and I want to move on since I have limited windows to work while on east coast... If you have bandwidth it would be super awesome to have:
attachment[0] types
attachment[0].url for images
Basically, it's going to look something like this:
If there are no attachments, show a preview of the post
If the post is an event, show the event start time as a prominent icon
If the post has an image attachment, show the image attachment and the title
if the post has an attachment other than image (type), show the icon for that type
This will make this view work a bit like google drive.

**Comment by brodeur** [comment]
Twerk time

File: src/components/PostGridItem/PostGridItem.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import cx from 'classnames'
+// import Moment from 'moment'
```
Comment (tibetsprague) [inline src/components/PostGridItem/PostGridItem.js:3]: remove

File: src/components/PostGridItem/PostGridItem.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    title,
+    details,
+    creator,
+    createdAt,
+    attachments
+  } = post
+
+  const firstAttachment = attachments[0] || {}
+  const { attachmentType, attachmentUrl } = firstAttachment
+  console.log(attachmentType, attachmentUrl, 'these should reliably be valid variables if they actually exist, or undefined.')
```
Comment (tibetsprague) [inline src/components/PostGridItem/PostGridItem.js:26]: console

File: src/components/PostGridItem/PostGridItem.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  if (!creator) { // PostCard guards against this, so it must be important? ;P
+    return null
+  }
+
+  const creatorUrl = personUrl(creator.id, routeParams.slug)
+  const unread = false
+  // will reintegrate once I have attachment vars
```
Comment (tibetsprague) [inline src/components/PostGridItem/PostGridItem.js:33]: @thomasgwatson ah looks like @brodeur is waiting on attachments

Comment (tibetsprague) [reply src/components/PostGridItem/PostGridItem.js:33]: or rather, event variables?

File: src/components/PostGridItem/PostGridItem.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  /* const startTimeMoment = Moment(post.startTime) */
+
+  return (
+    <div styleName={cx('post-grid-item-container', { unread, expanded })} onClick={showDetails}>
+      <div styleName='content-summary'>
+        <h3 styleName='title'>{title}</h3>
+
+        {/*  Will fix this after I get attachment variables */}
```
Comment (tibetsprague) [inline src/components/PostGridItem/PostGridItem.js:41]: @thomasgwatson

**Comment by thomasgwatson** [comment]
Yes, this is all waiting on another run by Aaron, and then we can zip it up

**Review by tibetsprague — APPROVED** [review]

**Pull request merged**